### ISSUE TITLE
Button label not shown on hover

### ISSUE THREAD

**Issue by mbohal** [react-ui-org/react-ui#489]
When the `<Button>` component is rendered with `labelVisbility` set to `none` or to value smaller the current viewport the `label` value is not accessible.

I think it should appear on mouse hover.

**Comment by adamkudrna** [comment]
The [invisible label](https://react-ui.io/components/button#icon-buttons) does remain accessible:

<img width="802" alt="obrazek" src="https://github.com/react-ui-org/react-ui/assets/5614085/f28fde9c-af2c-4698-843e-32b1d9e69260">

FF dev tools:
<img width="452" alt="obrazek" src="https://github.com/react-ui-org/react-ui/assets/5614085/57548afd-f2aa-42c9-9f2e-dcb4977fd580">

What you probably mean is the button does not provide any textual alternative to sighted users. In such case this is what you would do:

1. add the `title` attribute (which may be annoying for AT users as both texts are announced: _"My action, Title description, button"_, or something like that), or
2. combine the button with custom Tooltip; unfortunately, RUI currently does not have such component (Popover is close, but you need to click to make it show up).

To sum it up, current behavior is correct. But I'll be happy to implement a `Tooltip` component which could be paired with `Button`s to improve the UX while not putting accessibility at stake.

**Comment by mbohal** [comment]
> What you probably mean is the button does not provide any textual alternative to sighted users.

Yes, that is what I had in mind.

Lets go with option 1). When we will have the component `<Tooltip>` we can use it here. We do not have the capacity to implement `<Tooltip>` at this point.

**Comment by adamkudrna** [comment]
Then I suggest putting something like this into the docs:

_For icon-only buttons, consider using the `title` attribute to provide a textual alternative._

OK?

**Comment by mbohal** [comment]
@adamkudrna 

I think there are two ways to solve this:

1.  Update the docs as you suggest and make user pass in both `label` and `title` props
2.  Use user specified `title` if provided or use the text passed in via the `label`  prop if not provided

I think I prefer 2) developers (e.g. myself) will keep forgetting to pass in the `title` . Are there any arguments why 1) would be a better idea?

**Comment by mbohal** [comment]
The docs will be updated in the `General guidelines` section.

When mentioning the option to add `title` it should be mentioned that it will lead to aria reader a duplicit description.

**Comment by adamkudrna** [comment]
[Adobe Spectrum](https://spectrum.adobe.com/page/button/#Tooltip-when-the-label-is-hidden):

![Image](https://github.com/react-ui-org/react-ui/assets/5614085/f368c7bf-10ff-4ed3-a18d-36062decf93c)

… but seemingly, [they don't implement it](https://react-spectrum.adobe.com/react-spectrum/Button.html#icon-only):

![Image](https://github.com/react-ui-org/react-ui/assets/5614085/0c91ec90-297b-4654-9557-92955549d04c)

The reason probably is that in Spectrum they pass the button label as children so it's not easy/possible to find out if a label is rendered or not.
